manifests/instance.pp

```diff
@@ -129,13 +129,15 @@
     require => Archive["${instance_name}-${archive_name}"],
   }
 
-  exec { "${instance_name}-check-runner-configured":
-    user    => $user,
-    cwd     => "${github_actions_runner::root_dir}/${instance_name}",
-    command => 'true',
-    unless  => "test -f ${github_actions_runner::root_dir}/${instance_name}/runsvc.sh",
-    path    => ['/bin', '/usr/bin'],
-    notify  => Exec["${instance_name}-run_configure_install_runner.sh"],
+  if $ensure == 'present' {
```

Comment:
Could ensure have other typical values as "latest"? If that is the case maybe it would be better to check "!= 'absent'"

Reply:
ensure is defined as an Enum with just two values:
```
  Enum['present', 'absent']  $ensure                = 'present',
```
So it should be enough checking to be equal to present.